schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,370 @@
+        }
+      ]
+    },
+    "object_input_field_regex": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "regex": {
+          "type": "string"
+        },
+        "capture_group": {
+          "type": "integer"
+        },
+        "required": {
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "regex",
+        "capture_group"
+      ]
+    },
+    "object_input_field_env": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "env": {
+          "$ref": "#/definitions/input_env_variables"
+        },
+        "required": {
+          "type": "boolean"
+        }
+      },
+      "required": [
+        "env"
+      ]
+    },
+    "actions": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/connector_level_action"
+      }
+    },
+    "connector_level_action": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "url": {
+          "$ref": "#/definitions/link"
+        },
+        "user_input": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/connector_action_user_input"
+          }
+        },
+        "request": {
+          "type": "object",
+          "additionalProperties": {"$ref": "#/definitions/connector_action_request_params"}
+        },
+        "label": {
+          "type": "string"
+        },
+        "type": {
+          "$ref": "#/definitions/http_methods"
+        },
+        "action_key": {
+          "$ref": "#/definitions/connector_action_keys"
+        },
+        "metadata": {
+          "type": "object"
+        }
+      },
+      "if": {
+        "properties": {
+          "action_key": {
+            "const": "USER_INPUT"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "user_input"
+        ]
+      },
+      "else": {
+        "not": {
+          "required": [
+            "user_input"
+          ]
+        }
+      },
+      "required": [
+        "url",
+        "type",
+        "label",
+        "action_key"
+      ]
+    },
+    "connector_action_user_input": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "label": {
+          "type": "string"
+        },
+        "format": {
+          "$ref": "#/definitions/connector_action_user_input_formats"
+        },
+        "select": {
+          "type": "object"
+        },
+        "min_length": {
+          "type": "integer",
+          "minimum": 0
+        },
+        "max_length": {
+          "type": "integer",
+          "minimum": 0
+        }
+      },
+      "if": {
+        "properties": {
+          "format": {
+            "const": "select"
+          }
+        }
+      },
+      "then": {
+        "required": [
+          "select"
+        ]
+      },
```

Comment:
I missed one condition here, for "else" case.
` "else": {
        "not": {
          "required": ["select"]
        }
      }`

Reply:
Ignored. No need to be strict like this.

The other way is still enforced. That is, if the format is `select`, then the 'select' property is required.